editor/components/inner-blocks/index.js

```diff
@@ -28,9 +28,12 @@ class InnerBlocks extends Component {
 	}
 
 	componentDidUpdate( prevProps ) {
-		const { template, block } = this.props;
+		const { template, templateLock, block } = this.props;
 
 		this.updateNestedSettings();
+		if ( templateLock !== 'all' && block.innerBlocks && block.innerBlocks.length > 0 ) {
```

Comment:
We're checking truthiness of `block.innerBlocks` before testing `length` here but not below in the assignment of `isTemplateInnerBlockMismatch`. We should be consistent one way or the other, depending on whether we actually need to be defensive.